File: module-services/src/main/java/eu/geekhome/services/hardware/HardwarePlugin.java
package eu.geekhome.services.hardware;

import org.pf4j.Plugin;
import org.pf4j.PluginWrapper;

public abstract class HardwarePlugin extends Plugin {

    public HardwarePlugin(PluginWrapper wrapper) {
        super(wrapper);
    }

    public abstract HardwareAdapterFactory getFactory();
}

File: modules/domain/src/main/java/eu/geekhome/domain/hardware/HardwarePlugin.kt
package eu.geekhome.domain.hardware;

import org.pf4j.Plugin;
import org.pf4j.PluginWrapper;

public abstract class HardwarePlugin extends Plugin implements HardwareAdapterFactory {

    public HardwarePlugin(PluginWrapper wrapper) {
        super(wrapper);
    }
}
